File: a-java/ac/T799.java
package ac;

import java.util.*;

// 双指针 最长连续不重复子序列 https://www.acwing.com/problem/content/801/
class T799 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] q = new int[n];
        for (int i = 0; i < n; i++) {
            q[i] = sc.nextInt();
        }
            
        HashMap<Integer, Integer> m = new HashMap<>();
        int res = 0;
        for (int i = 0, j = 0; i < n; i++) {
            m.put(q[i], m.getOrDefault(q[i], 0) + 1);
            while (j < i && m.get(q[i]) > 1) {
                m.put(q[j], m.get(q[j]) - 1);
                j++;
            }
            res = Math.max(res, i - j + 1);
        }
        System.out.print(res);
        sc.close();
    }
}

File: b-go/ac/T799.go
package ac

import "fmt"

// 双指针 最长连续不重复子序列 https://www.acwing.com/problem/content/801/
func T790() {
    var n int
    fmt.Scan(&n)
    q := make([]int, n)
    for i := 0; i < n; i++ {
        fmt.Scan(&q[i])
    }
    m := make(map[int]int)
    res := 0
    for i, j := 0, 0; i < n; i++ {
        m[q[i]]++
        for j < i && m[q[i]] > 1 {
            m[q[j]]--
            j++
        }
        // math.Max()方法只能用于float64类型，需要自己实现
        if res < i - j + 1 {
            res = i - j + 1
        }
    }
    fmt.Print(res)
}